crates/milli/src/progress.rs

```diff
@@ -343,3 +391,16 @@ impl<T: steppe::Step> Step for Compat<T> {
         self.0.total().try_into().unwrap_or(u32::MAX)
     }
 }
+
+pub struct ScopedProgressStep<'a, P: Step + Copy> {
+    progress: &'a Progress,
+    step: Option<P>,
+}
+
+impl<'a, P: Step + Copy> Drop for ScopedProgressStep<'a, P> {
+    fn drop(&mut self) {
+        if let Some(step) = self.step {
+            self.progress.end_progress_step(step);
```

Comment:
`end_progress_step` may panic (for example, if the coder performed a manual progress.update while there is a ù ScopedProgressStep`), which will cause an abort in some cases as this is a destructor.

(panics in panicking destructors cause an immediate abort).